### ISSUE TITLE
Linux portable build

### ISSUE THREAD

**Issue by legluondunet** [CorsixTH/CorsixTH#1569]
Hello,

I would like to compile a portable Corsix-TH build, is it possible to compile dependencies statically?
I didn't find documentation about that.
Thank you for your help.

**Comment by TheCycoONE** [comment]
It is certainly possible, largely a matter of linking static libraries instead of dynamic - which means you'll likelt be compiling all the dependencies yourself. What OS?

**Comment by legluondunet** [comment]
Linux.

**Comment by legluondunet** [comment]
I can join libraries files and then use LD_LIBRARY_PATH ,this works but...I didn't find a way for LUA, do you know how to make this depency portable?

**Comment by TheCycoONE** [comment]
liblua.a should be the same. The lua flles themselves are data and cannot currently be built in. The lua executable dependency is only for build time so it doesn't matter.

**Comment by TheCycoONE** [comment]
lua_socket you won't need but you will need lfs. It can be anywhere in the lua cpath, so you could include it with the executable and still be 'portable' unless you are going for single file.

**Comment by TheCycoONE** [comment]
-DCORSIX_TH_INTERPRETER_PATH will also need to be set to the relative path to CorsixTH.lua

**Comment by legluondunet** [comment]
if I send you a package, do you have a Linux partition to test it? If so, which distribution?

**Comment by TheCycoONE** [comment]
All of my existing linux boxes and vms have the dependencies installed already. I could spin up a fresh vm, but so could you.

Are you suggesting to add this to the releases? If so, let me know what glibc version and architecture you are targeting and I will verify it.

**Comment by legluondunet** [comment]
I need help, I'm stuck here:


``
$ ./corsix-th.sh 
le répertoire courant est /home/legluondunet/Games/theme-hospital/Corsix-TH
An error has occurred in CorsixTH:
./Lua/strict.lua:66: module 'lfs' not found:
	no field package.preload['lfs']
	no file './Lua/lfs.lua'
	no file './Lua/lfs/init.lua'
	no file '/usr/local/share/lua/5.2/lfs.lua'
	no file '/usr/local/share/lua/5.2/lfs/init.lua'
	no file '/usr/local/lib/lua/5.2/lfs.lua'
	no file '/usr/local/lib/lua/5.2/lfs/init.lua'
	no file '/usr/share/lua/5.2/lfs.lua'
	no file '/usr/share/lua/5.2/lfs/init.lua'
	no file './lfs.lua'
	no file './lfs.so'
	no file '/usr/local/lib/lua/5.2/lfs.so'
	no file '/usr/lib/x86_64-linux-gnu/lua/5.2/lfs.so'
	no file '/usr/lib/lua/5.2/lfs.so'
	no file '/usr/local/lib/lua/5.2/loadall.so'
	no file './lfs.so'
stack traceback:
	[C]: in function 'fn'
	./Lua/strict.lua:66: in function 'require'
	./Lua/app.lua:23: in main chunk
	[C]: in function 'dofile'
	./CorsixTH.lua:72: in function 'require'
	./CorsixTH.lua:124: in main chunk
	[C]: in ?

```

infos: 
- compiled corsix-th on Ubuntu 18.04 x64
- all libs stored in libs folder came from Ubuntu 18.04
- to launch Corsix-th, use ./corsix-th.sh script (use LD_LIBRARY_PATH
- 64 bits only
- put all Theme Hospital assets in TH folder (optionnal)

How to add the missed LUA dependence's?  (LFS...)

Thank you for your help.

[theme-hospital.tar.gz](https://github.com/CorsixTH/CorsixTH/files/3926953/theme-hospital.tar.gz)

**Comment by legluondunet** [comment]
[corsixth_portable_porject_x64.zip](https://github.com/CorsixTH/CorsixTH/files/3928532/corsixth_portable_porject_x64.zip)

**Comment by legluondunet** [comment]
last revision 05/12/2019 20h07
[corsixth_portable_project_x64.zip](https://github.com/CorsixTH/CorsixTH/files/3928619/corsixth_portable_project_x64.zip)

**Comment by legluondunet** [comment]
last revision 06/12/2019 23h17
[corsixth_portable_project_x64.zip](https://github.com/CorsixTH/CorsixTH/files/3934230/corsixth_portable_project_x64.zip)

tested on Linux distribution fresh install:
Manjaro 18.1.3: libraries OK, Lua OK
Ubuntu 18.04: libraries OK, Lua OK

issues to resolve:
no music
DE integration

**Comment by TheCycoONE** [comment]
Great, I'll take a look. For midi, my notes on the aur might help: https://aur.archlinux.org/packages/corsix-th#comment-695085

**Comment by legluondunet** [comment]
I'm thinking about a way to make the "music part" portable, the most practice and easy way would be to use audio files (ogg, mp3, flac...).
Is there a way to convert midi files to audio files?

**Comment by TheCycoONE** [comment]
Yes, the game already supports mp3 and ogg vorbis (and flac) if you preconfigured the music directory. You cannot redistribute the original game music (copyright) but there is free music available in a similar spirit.

**Comment by TheCycoONE** [comment]
If you haven't already looked into it you may want to try static compiling everything with musl to avoid a glibc dependency.

**Comment by legluondunet** [comment]
ok but could you explain me how to do that?

**Comment by legluondunet** [comment]
last revision 09/12/2019 00h51

--> libraries OK, Lua OK, music OK, DE Integration OK
tested with success on Linux distributions fresh install:
Debian 10, Ubuntu 18.04, Manjaro 18.1.3, Solus 4, OpenSUSE Timbleweed and  Fedora 30.


[corsix-th_linux_x64_portable.zip](https://github.com/CorsixTH/CorsixTH/files/3937375/corsix-th_linux_x64_portable.zip)

**Comment by softbrix** [comment]
Hi @legluondunet !
Could you please share your build instructions so I can replicate those on my ARM build for RPi4 and debian:buster.

**Comment by legluondunet** [comment]
Hello @softbrix, all needed information are available in the archive:
- needed libraries
- bash script 
- compile options
...
If you need more precise info ask me, but my knowledge is limited with raspberry pi.

**Comment by lewri** [comment]
Is this one still intended to be open or is it resolved?

**Comment by colinjmatt** [comment]
I think we've only just confirmed this is now resolved, after #2147 is merged now that it's passed testing.
